MatrixSDK/JSONModels/MXEvent.m

```diff
@@ -574,6 +578,42 @@ - (NSArray *)readReceiptEventIds
     return list;
 }
 
+- (NSArray *)readReceiptThreadIds
+{
+    NSMutableArray* list = nil;
+    
+    if (_wireEventType == MXEventTypeReceipt)
+    {
+        NSArray* eventIds = [_wireContent allKeys];
+        list = [[NSMutableArray alloc] initWithCapacity:eventIds.count];
+        
+        for (NSString* eventId in eventIds)
+        {
+            NSDictionary* eventDict = [_wireContent objectForKey:eventId];
+            NSDictionary* readDict = [eventDict objectForKey:kMXEventTypeStringRead];
+            
+            if (readDict)
+            {
+                NSArray<NSDictionary *>* userDicts = [readDict allValues];
+                
+                NSString *threadId;
+                for (NSDictionary *userDict in userDicts)
+                {
+                    threadId = userDict[@"thread_id"];
+                    if (threadId)
+                    {
+                        break;
+                    }
+                }
+                
+                [list addObject:threadId ?: [NSNull null]];
```

Comment:
Why don't we use `kMXEventTimelineMain` instead of `NSNull`?

Reply:
We still must support RR without thread ID (unthreaded RR) sent by old clients. In this case we have to set the thread ID as `nil` so it the RR will be stored in all thread and main timelines as per MSC.